Makefile

```diff
@@ -93,82 +216,93 @@ SONG_OBJS := $(patsubst $(SONG_SUBDIR)/%.s,$(SONG_BUILDDIR)/%.o,$(SONG_SRCS))
 MID_SRCS := $(wildcard $(MID_SUBDIR)/*.mid)
 MID_OBJS := $(patsubst $(MID_SUBDIR)/%.mid,$(MID_BUILDDIR)/%.o,$(MID_SRCS))
 
-OBJS := $(C_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
+OBJS     := $(C_OBJS) $(C_ASM_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
 OBJS_REL := $(patsubst $(OBJ_DIR)/%,%,$(OBJS))
 
-TOOLDIRS := $(filter-out tools/agbcc tools/binutils,$(wildcard tools/*))
-TOOLBASE = $(TOOLDIRS:tools/%=%)
-TOOLS = $(foreach tool,$(TOOLBASE),tools/$(tool)/$(tool)$(EXE))
+SUBDIRS  := $(sort $(dir $(OBJS)))
+$(shell mkdir -p $(SUBDIRS))
 
-.PHONY: all rom tools clean-tools mostlyclean clean compare tidy speed_eaters unk_8D94B9C unk_8E1FE28 unk_8E8490C $(TOOLDIRS)
-
-MAKEFLAGS += --no-print-directory
-
-AUTO_GEN_TARGETS :=
-
-all: tools rom
+# Pretend rules that are actually flags defer to `make all`
+modern: all
+compare: all
 
+# Other rules
 rom: $(ROM)
 ifeq ($(COMPARE),1)
-	@$(SHA1) katam.sha1
+	@$(MAKE) -C $(SUBGAME_LOADERS) compare
+	@$(MAKE) -C multi_boot/unk_8D94B9C compare
+	@$(MAKE) -C multi_boot/unk_8E1FE28 compare
+	@$(MAKE) -C multi_boot/unk_8E8490C compare
+	@$(SHA1) $(FILE_NAME).sha1
```

Comment:
This is definitely broken. If you do `make clean && make compare -j`, the checksum of payload.gba is going to be compared 3 times.